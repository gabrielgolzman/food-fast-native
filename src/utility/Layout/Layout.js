import React from 'react';
import styled from 'styled-components/native';
import { IconButton, Button } from 'react-native-paper';

const Container = styled.View`
   flex: 1;
   flex-direction: column-reverse;
`;

const HeaderImage = styled.ImageBackground.attrs({
   source: require('../../../assets/img/header_restaurant.png'),
})`
   height: 30%;
   width: 100%;
   position: absolute;
   top: 0;
   display: flex;
   z-index: 1;
   flex-direction: row;
   justify-content: space-between;
`;

const SettingsButton = styled(IconButton).attrs({
   icon: 'cog',
   size: 35,
   color: 'orange',
})`
   z-index: 10;
`;

const HelpButton = styled(Button).attrs({
   icon: 'bell-ring',
   color: 'orange',
   mode: 'contained',
   labelStyle: {
      fontSize: 13,
      color: 'white',
   },
})`
   border-radius: 20px;
   font-size: 10px;
   justify-content: center;
   width: 70%;
   z-index: 10;
   margin: 10px;
   height: 40px;
`;

const Main = styled.View`
   height: 70%;
`;

const Layout = ({ children }) => {
   return (
      <Container>
         <HeaderImage>
            <SettingsButton
               onPress={() => console.log('Pressed the settings!')}
            />
            <HelpButton onPress={() => console.log('Help needed!')}>
               Solicitar ayuda al personal
            </HelpButton>
         </HeaderImage>
         <Main>{children}</Main>
      </Container>
   );
};

export default Layout;
