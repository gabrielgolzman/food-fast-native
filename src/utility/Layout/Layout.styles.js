import styled from 'styled-components/native';
import { IconButton } from 'react-native-paper';

export const Container = styled.View`
   flex: 1;
   flex-direction: column-reverse;
`;

export const HeaderImage = styled.ImageBackground.attrs({
   source: require('../../../assets/img/header_restaurant.png'),
})`
   height: 30%;
   width: 100%;
   position: absolute;
   top: 0;
   display: flex;
   flex-direction: column;
   z-index: 1;
   justify-content: space-between;
`;

export const TopButtonsContainer = styled.View`
   width: 100%;
   display: flex;
   flex-direction: row;
   justify-content: space-between;
`;

export const SettingsButton = styled(IconButton).attrs((props) => ({
   icon: 'cog',
   size: 35,
   color: props.theme.colors.ui.primary,
}))`
   z-index: 10;
`;

export const Main = styled.View`
   height: 70%;
   justify-content: flex-end;
   align-items: center;
`;
