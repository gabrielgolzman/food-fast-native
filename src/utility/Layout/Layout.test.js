import React from 'react';
import { render } from '@testing-library/react-native';

import Layout from './Layout';

it('renders the Layout correctly', () => {
   const tree = render(<Layout />).toJSON();
   expect(tree).toMatchSnapshot();
});
