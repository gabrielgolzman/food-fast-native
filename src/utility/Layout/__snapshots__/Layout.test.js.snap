// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Layout correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexDirection": "column-reverse",
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "row",
          "height": "30%",
          "justifyContent": "space-between",
          "position": "absolute",
          "top": 0,
          "width": "100%",
          "zIndex": 1,
        },
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": undefined,
        }
      }
      accessible={true}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 6,
          "left": 6,
          "right": 6,
          "top": 6,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "margin": 6,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 26.25,
              "height": 52.5,
              "width": 52.5,
            },
            undefined,
            Array [
              Object {
                "zIndex": 10,
              },
            ],
          ],
        ]
      }
    >
      <View>
        <Text />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "orange",
          "borderColor": "transparent",
          "borderRadius": 20,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 2,
          "fontSize": 10,
          "height": 40,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
          "minWidth": 64,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.75,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 1.5,
          "width": "70%",
          "zIndex": 10,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 20,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "marginLeft": 12,
                "marginRight": -4,
              }
            }
          >
            <Text />
          </View>
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "color": "white",
                    "fontSize": 13,
                  },
                ],
              ]
            }
          >
            Solicitar ayuda al personal
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "height": "70%",
        },
      ]
    }
  />
</View>
`;
